### ISSUE TITLE
Have guardian add comment to PR when apply fails due to plan being more than 7 days old

### ISSUE THREAD

**Issue by kculligan** [abcxyz/guardian#416]
### TL;DR

Have guardian add comment to PR when apply fails due to plan being more than 7 days old. Currently in this scenario the terraform apply fails but there's no notification in the PR that something went wrong

### Detailed design

```markdown

```

### Alternatives considered

```markdown

```

### Additional information

Scenario:

1. PR is created, all checks passed (including guardian)
2. More than 7 days later the PR is reviewed and merged
3. Terraform apply fails to due the plan no longer existing
4. No notification in the PR comment thread
5. Developer at some point notices that there's an issue and has to troubleshoot (maybe seeing that a workflow failed)
6. Developer pushes a noop commit to regenerate the plan and then merges the change.

Desired behavior: Guardian adds a comment to the PR to notify developers that something went wrong (ideally suggesting the source of the issue)

If needed we can provided a PR + Workflow examples in a non-public repo.
